ansible_events_ui/api/activation.py

```diff
@@ -1,8 +1,125 @@
 """Activation API endpoints."""
-from fastapi import APIRouter
+import logging
+import json
+
+from fastapi import APIRouter, Depends
+import asyncio
+
+import sqlalchemy.orm
+from sqlalchemy import insert, select
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from ansible_events_ui.db.dependency import (
+    get_db_session,
+    get_db_session_factory,
+)
+from ansible_events_ui.db.models import (
+    activations,
+    extra_vars,
+    inventories,
+    rule_set_files,
+    activation_instance_logs,
+    playbooks,
+    execution_envs,
+)
+from ansible_events_ui.managers import (
+    taskmanager,
+    connectionmanager,
+)
+from ansible_events_ui.ruleset import (
+    activate_rulesets,
+)
+from ansible_events_ui.schemas import (
+    Activation,
+)
+
+logger = logging.getLogger("ansible_events_ui")
 
 __all__ = ("router",)
 
 router = APIRouter()
 
-# Implement activation API here.
+@router.post("/api/activation/")
+async def create_activation(
+    a: Activation,
+    db: AsyncSession = Depends(get_db_session),
+    db_session_factory: sqlalchemy.orm.sessionmaker = Depends(
+        get_db_session_factory
+    ),
+):
+    query = select(rule_set_files).where(
+        rule_set_files.c.id == a.rule_set_file_id
+    )
+    rule_set_file_row = (await db.execute(query)).first()
+
+    query = select(inventories).where(inventories.c.id == a.inventory_id)
+    inventory_row = (await db.execute(query)).first()
+
+    query = select(extra_vars).where(extra_vars.c.id == a.extra_var_id)
+    extra_var_row = (await db.execute(query)).first()
+
+    query = select(playbooks).where(playbooks.c.id == a.playbook_id)
+    playbook_row = (await db.execute(query)).first()
+
+    query = select(execution_envs).where(execution_envs.c.id == a.execution_env_id)
+    execution_env_row = (await db.execute(query)).first() 
+
+    query = insert(activations).values(
+        name=a.name,
+        description=a.description,
+        rule_set_file_id=a.rule_set_file_id,
+        inventory_id=a.inventory_id,
+        execution_env_id=a.execution_env_id,
+        restart_policy_id=a.restart_policy_id,
+        playbook_id=a.playbook_id,
+        activation_enabled=a.activation_enabled,
+        extra_var_id=a.extra_var_id,
+        created=a.created,
+        last_modified=a.last_modified
+    )
+    result = await db.execute(query)
+    await db.commit()
+    (id_,) = result.inserted_primary_key
+
+    cmd, proc = await activate_rulesets(
```

Comment:
Since the activation represents a template of actual execution of an `ansible-events` process, when activation is created, an `ansible-events` process probably should not be spawned. But if it were to spawn it, an activation instance should be created.

@benthomasson How an activation is going to be triggered? Will there be a separate API call for that?